src/pages/steps/AdditionalInformationStep/ContactInfoStep.tsx

```diff
@@ -112,17 +115,53 @@ const ContactInfoStep = ({
     });
 
     setContactInfoState(contactInfoArray);
-  }, [contactInfo, onValidationChange]);
+    setIsContactInfoInitialized(true);
+  }, [contactInfo]);
+
+  useEffect(() => {
+    if (!isContactInfoStateInitialized) return;
+
+    const filteredContactInfoState = contactInfoState.filter(
+      (contactInfo) => contactInfo.value !== '',
+    );
+
+    if (filteredContactInfoState.length === 0) {
+      onValidationChange(ValidationStatus.NONE);
+      return;
+    }
+
+    onValidationChange(ValidationStatus.SUCCESS);
+  }, [contactInfoState]);
+
+  const queryClient = useQueryClient();
 
   const addContactPointMutation = useAddOfferContactPointMutation({
+    onMutate: async (newPayload) => {
+      await queryClient.cancelQueries({
+        queryKey: [scope, { id: offerId }],
+      });
+
+      const previousEventInfo: any = queryClient.getQueryData([
+        scope,
+        { id: offerId },
+      ]);
+
+      return { previousEventInfo };
+    },
+    onError: (_err, _newBookingInfo, context) => {
+      queryClient.setQueryData(
+        [scope, { id: offerId }],
+        context.previousEventInfo,
+      );
+    },
     onSuccess: onSuccessfulChange,
   });
 
   const parseNewContactInfo = (newContactInfo: NewContactInfo[]) => {
     const [email, phone, url] = Object.values(ContactInfoTypes).map(
       (infoType) =>
         newContactInfo
-          .filter((info) => info.type === infoType)
+          .filter((info) => info.type === infoType && info.value !== '')
```

Comment:
```suggestion
          .filter((info) => info.type === infoType && info.value !== '')
```

Should this be `&&` or `||`? 
Is it only the empty string values of infoType that you want to filter? 
Or do you want to filter out empty strings in general?

Reply:
👍🏻